WeatherMonitoringAndReportingService/InputConversion/XMLToWeatherDetailsAdapter.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using WeatherMonitoringAndReportingService.WeatherDetails;
+
+namespace WeatherMonitoringAndReportingService.InputConversion;
+
+public class XMLToWeatherDetailsAdapter : IAdapter
+{
+    public static WeatherDetailsModel ToWeatherDetailsAdapter(string xml)
+    {
+        XmlSerializer serializer = new XmlSerializer(typeof(WeatherDetailsModel), new XmlRootAttribute("WeatherData"));
+        using (StringReader reader = new StringReader(xml))
+        {
+            return (WeatherDetailsModel)serializer.Deserialize(reader);
+        }
+    }
+}
```

Comment:
I would declare a const for the RootAttribute, also try to use var whenever the type is explicit from the right side. 

Simplify the using.

```csharp
public class XMLToWeatherDetailsAdapter : IAdapter
{
    private const string RootAttribute = "WeatherData";
    
    public static WeatherDetailsModel ToWeatherDetailsAdapter(string xml)
    {
        var serializer = new XmlSerializer(typeof(WeatherDetailsModel), new XmlRootAttribute(RootAttribute));
        var reader = new StringReader(xml);
        return (WeatherDetailsModel)serializer.Deserialize(reader);
    }
}
```